cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -110,3 +160,37 @@ func (pe *ProbeEvent) initOrErrorEvent() {
 		klog.Error(err)
 	}
 }
+
+// generateUUID creates a new UUID based on the algorithm proposed in
+// https://github.com/openebs/openebs/pull/2666
+func generateUUID(bd blockdevice.BlockDevice) (string, bool) {
+	var ok bool
+	var uuidField, uuid string
+
+	// select the field which is to be used for generating UUID
+	switch {
+	case bd.DeviceAttributes.DeviceType == libudevwrapper.UDEV_PARTITION:
+		klog.Infof("device(%s) is a partition, using partition UUID: %s", bd.DevPath, bd.PartitionInfo.PartitionEntryUUID)
+		uuidField = bd.PartitionInfo.PartitionEntryUUID
+		ok = true
+	case len(bd.DeviceAttributes.WWN) > 0:
+		klog.Infof("device(%s) has a WWN, using WWN: %s", bd.DevPath, bd.DeviceAttributes.WWN)
+		uuidField = bd.DeviceAttributes.WWN
+		ok = true
+	case len(bd.PartitionInfo.PartitionTableUUID) > 0:
+		klog.Infof("device(%s) has a partition table, using partition table UUID: %s", bd.DevPath, bd.PartitionInfo.PartitionTableUUID)
+		uuidField = bd.PartitionInfo.PartitionTableUUID
+		ok = true
+	case len(bd.FSInfo.FileSystemUUID) > 0:
```

Comment:
Should there be a case of adding serial number also to this list or is serial number not a unique field?

Reply:
Serial number is not unique in some cloud environments. For example, in GCP the serial number is configurable by the `--device-name` flag while attaching the disk. If this flag is not provided, GCP automatically assigns the serial number which is unique only to the node.Hence we cannot use serial number.

Reply:
Probably include the above information also as a comment. It will be helpful for future refactoring.

Reply:
Done.